<!DOCTYPE html>
<html lang="en">
<head>
<meta charset="UTF-8">
<!--[if IE]><meta http-equiv="X-UA-Compatible" content="IE=edge"><![endif]-->
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="generator" content="Asciidoctor 1.5.2">
<title>Plugin fill-tags</title>
<link rel="stylesheet" href="./index.css">
</head>
<body class="article">
<div id="header">
</div>
<div id="content">
<div class="sidebarblock navig">
<div class="content">
<div class="ulist">
<div class="title">General</div>
<ul>
<li>
<p><a href="index.html">Main page</a></p>
</li>
<li>
<p><a href="../bcftools.html">Manual page</a></p>
</li>
<li>
<p><a href="install.html">Installation</a></p>
</li>
</ul>
</div>
<div class="ulist">
<div class="title">Calling</div>
<ul>
<li>
<p><a href="cnv-calling.html">CNV calling</a></p>
</li>
<li>
<p><a href="csq-calling.html">Consequence calling</a></p>
</li>
<li>
<p><a href="consensus-sequence.html">Consensus calling</a></p>
</li>
<li>
<p><a href="roh-calling.html">ROH calling</a></p>
</li>
<li>
<p><a href="variant-calling.html">Variant calling and filtering</a></p>
</li>
</ul>
</div>
<div class="ulist">
<div class="title">Tips and Tricks</div>
<ul>
<li>
<p><a href="convert.html">Converting formats</a></p>
</li>
<li>
<p><a href="query.html">Extracting information</a></p>
</li>
<li>
<p><a href="filtering.html">Filtering expressions</a></p>
</li>
<li>
<p><a href="plugins.html">Plugins</a></p>
</li>
</ul>
</div>
</div>
</div>
<div id="main">
<div class="sect1">
<h2 id="_plugin_fill_tags">Plugin fill-tags</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This plugin can be used to compute and fill various INFO tags:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>AN .. Total number of alleles in called genotypes</p>
</li>
<li>
<p>AC .. Allele count in genotypes</p>
</li>
<li>
<p>NS .. Number of samples with data</p>
</li>
<li>
<p>AC_Hom .. Allele counts in homozygous genotypes</p>
</li>
<li>
<p>AC_Het .. Allele counts in heterozygous genotypes</p>
</li>
<li>
<p>AC_Hemi .. Allele counts in hemizygous genotypes (haploid)</p>
</li>
<li>
<p>AF .. Allele frequency</p>
</li>
<li>
<p>MAF .. Minor Allele frequency</p>
</li>
<li>
<p>HWE .. HWE test (PMID:15789306)</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>By default the values are calculated across all samples, but also
per-population values can be calculated. For this,
provide a file with the list of samples in the first column and
a comma-separated list of populations in the second column. The
file can look for example like this:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>Sample1     Group1
Sample2     Group1,Group2
Sample3     Group2</pre>
</div>
</div>
<div class="paragraph">
<p>The list of plugin-specific options can be obtained by running
<code>bcftools +fill-tags -h</code>, which will print the following usage page:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>About: Set INFO tags AF, AC, AC_Hemi, AC_Hom, AC_Het, AN, HWE, MAF, NS.
Usage: bcftools +fill-tags [General Options] -- [Plugin Options]
Options:
   run "bcftools plugin" for a list of common options

Plugin options:
   -d, --drop-missing          do not count half-missing genotypes "./1" as hemizygous
   -t, --tags LIST             list of output tags. By default, all tags are filled.
   -S, --samples-file FILE     list of samples (first column) and comma-separated list of populations (second column)

Example:
   bcftools +fill-tags in.bcf -Ob -o out.bcf
   bcftools +fill-tags in.bcf -Ob -o out.bcf -- -t AN,AC
   bcftools +fill-tags in.bcf -Ob -o out.bcf -- -d
   bcftools +fill-tags in.bcf -Ob -o out.bcf -- -S sample-group.txt -t HWE</pre>
</div>
</div>
<div class="sect2">
<h3 id="_feedback">Feedback</h3>
<div class="paragraph">
<p>We welcome your feedback, please help us improve this page by
either opening an issue on github or editing it directly and sending
a pull request.</p>
</div>
</div>
</div>
</div>
</div>
</div>
<div id="footer">
<div id="footer-text">
</div>
</div>
</body>
</html>